src/compiler/types.ts

```diff
@@ -3674,6 +3674,8 @@ namespace ts {
         ContainsAnyFunctionType = 1 << 26,  // Type is or contains the anyFunctionType
         NonPrimitive            = 1 << 27,  // intrinsic object type
         /* @internal */
+        UnionOfUnitTypes        = 1 << 28,  // Type is union of unit types

```

Comment:
With this, `TypeFlags` is full. We're going to need a new field for flags like this (very) soon (as I don't see any that are obviously `ObjectFlags`).

Reply:
Well, `TypeFlags` being full is nothing new. Seems like it has been for the last many years. I actually think it is fine to put all of the flags to good use and then work out alternate solutions as the need arises. This particular flag could easily be moved to a property local to union types, but there's no reason to do so now.